### PR TITLE
upgrade storybook to 5.1.11

### LINKED ISSUE #397: Migrate to newest version of Storybook
# Feature request
**Is your feature request related to a problem? Please describe.**
#247 has been open for a long time.

**Describe the solution you'd like**
Bump Storybook and relevant add-ons to latest version and resolve any bugs that show up in `yarn storybook` and `yarn build-storybook`. Change whatever is needed to get it to work again!

### PR DESCRIPTION
**Pull request by chynh**
# Description of changes
Migrate to Storybook 5.1.11 
Minor changes to jest tests to remove console warnings

# Issue Resolved

Fixes #397

### CODE REVIEW

File: .storybook/webpack.config.js [hunk 1]
```diff
@@ -44,20 +58,21 @@ module.exports = (storybookBaseConfig, configType) => {
...
         },
       ],
+      include: path.resolve(__dirname, '../'),
     },
     {
       test: /\.svg$/,
       use: [
         {
-          loader: '@svgr/webpack',
+          loader: 'react-svg-loader',
```
Comment (kylemh) [inline .storybook/webpack.config.js:67]: What was the reasoning for this change?

**Comment by kylemh** [comment]
Just a reminder @chynh - you're a member so you don't need to fork this repo. You can just make branches directly off this repo. It'll be beneficial for me so I don't have to use https://hub.github.com/ to checkout your PR

File: .storybook/webpack.config.js (hunk 1 above)
Comment (chynh) [reply .storybook/webpack.config.js:67]: good point. I think I was following along the progress made in the other PR. Looks like it's not really necessary as I see it is working without this change. Should I go ahead and remove `react-svg-loader`?

**Comment by chynh** [comment]
> Just a reminder @chynh - you're a member so you don't need to fork this repo. You can just make branches directly off this repo. It'll be beneficial for me so I don't have to use https://hub.github.com/ to checkout your PR

Oh yeah I did not know this. Thanks for letting me know.

> Anyways, the Netlify build seems to be failing and not because of caching issues.

Seems like you guys had the same issue before. Were you able to resolve it? 
Maybe it has something to do with this message. `Module not found: Error: Can't resolve 'fs' in '/opt/build/repo/node_modules/require-context.macro'`

**Comment by kylemh** [comment]
😅  encountering my own demons

https://github.com/storybookjs/require-context.macro/issues/25

A few ideas:
1. Add `fs` as a `devDependency`. I don't think this will work, but it'd be the quickest fix.
2. Go to Storybook 5.3.x and use `.storybook/main.js` instead of `.storybook/{addons|config|backgrounds}.js` [since this fs/webpack interop doesn't seem to be required anymore](https://github.com/storybookjs/storybook/tree/master/addons/storyshots/storyshots-core#configure-jest-to-work-with-webpacks-requirecontext).
3. Use the hook instead of the macro https://github.com/storybookjs/storybook/tree/v5.1.11/addons/storyshots/storyshots-core#option-1-plugin
4. I want to move away from Netlify cuz it's costing me $5/mo. Perhaps Vercel has a better container environment that "just works" :tm:. You could add a 2nd build to `now.json` (I'm just worried about what this would mean for our GitHub PR integration tho)

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 54d41eb4968db2e95eea95efcad1918e5cc3fc10

https://deploy-preview-1132--operation-code-storybook.netlify.app

**Comment by kylemh** [comment]
Getting `TypeError: fs.readdirSync is not a function` in the runtime on the Netlify deploy

**Comment by chynh** [comment]
> Getting `TypeError: fs.readdirSync is not a function` in the runtime on the Netlify deploy

hmm yeah I saw that. I guess I'll have to spend some time looking into this issue this weekend. Is it possible for me to trigger netlify deploy manually?

**Comment by kylemh** [comment]
You're more than welcome to make your own Netlify account and deploy your Storybook builds willy nilly!

**Comment by chynh** [comment]
I was able to deploy without the weird 'fs' issue on my netlify account. Not sure what is causing issue on your account. Maybe the build setting?

Below is my setting:
Screen Shot 2020-06-14 at 2 03 56 PM

Ideally, it would be great to add 2nd build to Vercel but I am not sure how to do that. Been looking into document but not have been successful so far.  ¯\\\_(ツ)_/¯

**Comment by kylemh** [comment]
That was super helpful thanks! I had `NODE_ENV` and `BABEL_ENV` set to `test` in the environment variables. I got rid of those and we're flying! Woohoo!

**Comment by kylemh** [comment]
What was the reasoning for going specifically to version 5.1.11 as opposed to something more recent?

**Comment by chynh** [comment]
> What was the reasoning for going specifically to version 5.1.11 as opposed to something more recent?

I initially tried version 5.2.x but had issue with Accordion component. I don’t know what changed but Storybook seemed to have issue with passing JSX as prop. Might be something to keep in mind when upgrading in the future.

**Pull request merged**